packages/ag-charts-community/src/chart/series/cartesian/lineInterpolation.ts

```diff
@@ -0,0 +1,502 @@
+            ];
+        case 'cubic':
+            return [
+                { x: span.cp0x, y: span.cp0y },
+                { x: span.cp3x, y: span.cp3y },
+            ];
+    }
+}
+
+export function reverseSpan(span: Span): Span {
+    switch (span.type) {
+        case 'linear':
+            return {
+                type: 'linear',
+                moveTo: span.moveTo,
+                x0: span.x1,
+                y0: span.y1,
+                x1: span.x0,
+                y1: span.y0,
+            };
+        case 'cubic':
+            return {
+                type: 'cubic',
+                moveTo: span.moveTo,
+                cp0x: span.cp3x,
+                cp0y: span.cp3y,
+                cp1x: span.cp2x,
+                cp1y: span.cp2y,
+                cp2x: span.cp1x,
+                cp2y: span.cp1y,
+                cp3x: span.cp0x,
+                cp3y: span.cp0y,
+            };
+        case 'step':
+            return {
+                type: 'step',
+                moveTo: span.moveTo,
+                x0: span.x1,
+                y0: span.y1,
+                x1: span.x0,
+                y1: span.y0,
+                stepX: span.stepX,
+            };
+    }
+}
+
+export function rescaleSpan(span: Span, nextStart: Point, nextEnd: Point): Span {
+    const [prevStart, prevEnd] = spanRange(span);
+    const widthScale = prevEnd.x !== prevStart.x ? (nextEnd.x - nextStart.x) / (prevEnd.x - prevStart.x) : 0;
+    const heightScale = prevEnd.y !== prevStart.y ? (nextEnd.y - nextStart.y) / (prevEnd.y - prevStart.y) : 0;
+
+    switch (span.type) {
+        case 'linear':
+            return {
+                type: 'linear',
+                moveTo: span.moveTo,
+                x0: nextStart.x,
+                y0: nextStart.y,
+                x1: nextEnd.x,
+                y1: nextEnd.y,
+            };
+        case 'cubic':
+            return {
+                type: 'cubic',
+                moveTo: span.moveTo,
+                cp0x: nextStart.x,
+                cp0y: nextStart.y,
+                cp1x: nextEnd.x - (span.cp2x - prevStart.x) * widthScale,
+                cp1y: nextEnd.y - (span.cp2y - prevStart.y) * heightScale,
+                cp2x: nextEnd.x - (span.cp1x - prevStart.x) * widthScale,
+                cp2y: nextEnd.y - (span.cp1y - prevStart.y) * heightScale,
+                cp3x: nextEnd.x,
+                cp3y: nextEnd.y,
+            };
+        case 'step':
+            return {
+                type: 'step',
+                moveTo: span.moveTo,
+                x0: nextStart.x,
+                y0: nextStart.y,
+                x1: nextEnd.x,
+                y1: nextEnd.y,
+                stepX: nextEnd.x - (span.stepX - prevStart.x) * widthScale,
+            };
+    }
+}
+
+function setMoveTo(span: Span, moveTo: boolean) {
+    return span.moveTo !== moveTo ? { ...span, moveTo } : span;
+}
+
+export function splitSpanAtX(span: Span, x: number): [Span, Span] {
```

Comment:
I'm not using this - but I'll leave it in just in case it's needed for the line series